### PR TITLE
chore: add test for `MockKeyValueStore`

### LINKED ISSUE #1986: Test `MockKeyValueStorage`
# Motivation

The `MockKeyValueStorage` allows writing unit tests that use a `KeyValueStorage` without having to interface with a real storage layer. It should be tested to ensure it behaves correctly.

# Proposal

Write unit tests that use `KeyValueStorage::mock()` and it's `.to_mut()` variant, in order to also cover the proxy code in `KeyValueStorage` to call `MockKeyValueStorage`.

### PR DESCRIPTION
**Pull request by duguorong009**
## Motivation
- Resolve #1986 

## Proposal
- Add the unit test for `MockKeyValueStore`

## Test Plan
CI

### CODE REVIEW

**Comment by duguorong009** [comment]
@jvff 
Could you review the PR?

File: linera-sdk/Cargo.toml [hunk 1]
```diff
@@ -48,6 +48,10 @@ serde.workspace = true
 serde_json.workspace = true
 wit-bindgen.workspace = true
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+anyhow.workspace = true
+tokio = { workspace = true, features = ["rt", "macros"] }
```
Comment (jvff) [inline linera-sdk/Cargo.toml:53]: These dependencies should not be included when building for Wasm. You can instead mark the test to only run on non-Wasm targets, using `#[cfg(not(target_arch = "wasm"))]`.

File: linera-sdk/src/views/system_api.rs [hunk 2]
```diff
@@ -346,3 +346,39 @@ impl From<KeyValueStore> for ViewStorageContext {
...
+    assert!(is_key_existing);
+
+    let value = mock_store.read_value(b"foo").await?;
+    assert_eq!(value, Some(32_u128));
+
+    let value = mock_store.read_value(b"bar").await?;
+    assert_eq!(value, Some(42_u128));
+
+    Ok(())
+}
```
Comment (jvff) [inline linera-sdk/src/views/system_api.rs:384]: Two small comments:

1. We usually place the tests in a separate module
2. `with_testing` would build the test even when it doesn't need to run, so it's best to use `#[cfg(test)]` instead

```suggestion
#[cfg(all(test, not(target_arch = "wasm32")))]
mod tests {
    use super::*;

    #[tokio::test]
    async fn test_key_value_store_mock() -> anyhow::Result<()> {
        // Create a mock key-value store for testing
        let store = KeyValueStore::mock();
        let mock_store = store.to_mut();

        // Check if key exists
        let is_key_existing = mock_store.contains_key(b"foo").await?;
        assert!(!is_key_existing);

        // Check if keys exist
        let is_keys_existing = mock_store
            .contains_keys(vec![b"foo".to_vec(), b"bar".to_vec()])
            .await?;
        assert!(!is_keys_existing[0]);
        assert!(!is_keys_existing[1]);

        // Read and write values
        let mut batch = Batch::new();
        batch.put_key_value(b"foo".to_vec(), &32_u128)?;
        batch.put_key_value(b"bar".to_vec(), &42_u128)?;
        mock_store.write_batch(batch).await?;

        let is_key_existing = mock_store.contains_key(b"foo").await?;
        assert!(is_key_existing);

        let value = mock_store.read_value(b"foo").await?;
        assert_eq!(value, Some(32_u128));

        let value = mock_store.read_value(b"bar").await?;
        assert_eq!(value, Some(42_u128));

        Ok(())
    }
}
```

**Review by jvff — CHANGES_REQUESTED** [review]

File: linera-sdk/src/views/system_api.rs (hunk 2 above)
Comment (duguorong009) [reply linera-sdk/src/views/system_api.rs]: Thx for help! @jvff 
Done e09f125

**Review by jvff — APPROVED** [review]

**Pull request merged**